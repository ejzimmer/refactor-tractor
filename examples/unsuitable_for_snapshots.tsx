import React from 'react'
import { render } from '@testing-library/react'

interface Props {
  onConfirm: () => void
  onCancel: () => void
}

export function ConfirmationModal({ onConfirm, onCancel }: Props) {

  const handleCancel = (event: Event) => {
    event.preventDefault();
    onCancel();
  }

  const handleSpace = (event: KeyboardEvent) => {
    const keyPressed = event.code
    if (keyPressed === 'Space') {
      handleCancel(event);
    }
  }

  return (
    <div>
      <p>Can we fix it?</p>
      <div>
        <button onClick={onConfirm}>Yes, we can!</button>
        <a onClick={handleCancel} onKeyDown={handleSpace} href="#">No, we can't</a>
      </div>
    </div>
  )
}

describe('ConfirmationModal', () => {
  it('matches the snapshot', () => {
    const { container } = render(<ConfirmationModal onConfirm={jest.fn()} onCancel={jest.fn()} />)

    expect(container).toMatchSnapshot()
  })
})