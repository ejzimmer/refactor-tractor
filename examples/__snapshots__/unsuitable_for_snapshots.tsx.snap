// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal matches the snapshot 1`] = `
<div>
  <div>
    <p>
      Can we fix it?
    </p>
    <div>
      <button>
        Yes, we can!
      </button>
      <a
        href="#"
      >
        No, we can't
      </a>
    </div>
  </div>
</div>
`;
